docs/APIContract.md

```diff
@@ -0,0 +1,67 @@
+Furthermore this may allow having custom endpoints for the API.
+
+| Feature          | Field         | Type | Description                                                                       |
+| ---------------- | ------------- | ---- | --------------------------------------------------------------------------------- |
+| League Filtering | league_filter | bool | This can  be used to indicate whether the API supports filtering based on leagues |
+| Gem Filtering    | gem_filter    | bool | This can  be used to indicate whether the API supports filtering based on gems    |
+
+## Version 1
+
+### Response and Request types
+
+#### BuildInfo
+
+| Field        | Type    | Description                         |
+| ------------ | ------- | ----------------------------------- |
+| pobdata      | string  | The Path of Building data           |
+| name         | string  | The name of the build               |
+| lastModified | integer | The last modification timestamp     |
+| buildId      | string  | The unique identifier for the build |
+
+### Endpoints
+
+<details>
+ <summary><code>GET</code> <code><b>/v1/builds</b></code> <code>(Lists multiple builds)</code></summary>
+
+#### Parameters
+> Query Parameters
+
+| name   | type     | data type | description                             |
+| ------ | -------- | --------- | --------------------------------------- |
+| page   | optional | integer   | Used for pagination                     |
+| league | optional | string    | Used to limit builds to a league        |
+| gems   | optional | string    | Used to limit builds with specific gems |
+
+##### League
+
+The values should be matching what Grinding Gear Games will be using for the teaser part of the website, such as: _https://www.pathofexile.com/settlers_ or _https://www.pathofexile.com/affliction_. This allows for easier mapping of the data as neither PoB nor the API will be required to wait for either party.
+
+These links are generally available via poewiki see: https://www.poewiki.net/wiki/Necropolis_league _Official Page_.
+
+Example values:
+
+| Patch | League                 | value      | url                                    |
+| ----- | ---------------------- | ---------- | -------------------------------------- |
+| 3.25  | Settlers of Kalguur    | settlers   | https://www.pathofexile.com/settlers   |
+| 3.24  | Necropolis             | necropolis | https://www.pathofexile.com/necropolis |
+| 3.23  | Affliction             | affliction | https://www.pathofexile.com/affliction |
+| 3.22  | Trial of the Ancestors | ancestor   | https://www.pathofexile.com/ancestor   |
+| 3.4   | Delve                  | delve      | https://www.pathofexile.com/delve      |
+
+#### Responses
+
+| http code | content-type       | response           |
+| --------- | ------------------ | ------------------ |
+| `200`     | `application/json` | `BuildInfo object` |
```

Comment:
We probably need more metadata, for example a description/name of the list returned (or multiple?). pobarchives may provide a list of recent builds, pobbin would only provide popular.

This need goes away if every 'provider' can only provide a single list, but currently more is possible.

Reply:
So possibly a wrapper object such as:

```json
{
  "totalPages": 100,
  "buildInfos": [{}],
}```

something like this could then be used for assumption based "this is the last page".
Alas I'm still thinking about how multiple sources should be displayed together as in if every source has 10 entries and I have 100 sources I would get 1000 builds in technically one PoB page (ignoring that local pagination simulation is possible).